src/Simplify.elm

```diff
@@ -5156,62 +5163,59 @@ listIntersperseCompositionChecks checkInfo =
-                                    { arg = listArgToTheLeft
-                                    , argRepresents = "first list argument"
-                                    }
-                                    checkInfo
-                                )
+                Just secondArg_ ->
+                    if collection.empty.is checkInfo.lookupTable secondArg_ then
+                        Just
+                            (Rule.errorWithFix
+                                { message = "Unnecessary " ++ Tuple.second checkInfo.fn ++ " with " ++ descriptionForIndefinite collection.empty.description
+                                , details = [ "You can replace this call by the " ++ collection.represents ++ " itself." ]
+                                }
+                                checkInfo.fnRange
+                                (keepOnlyFix { parentRange = checkInfo.parentRange, keep = Node.range checkInfo.firstArg })
+                            )
 
-                        Just (_ :: _) ->
-                            case AstHelpers.getListLiteral listArgToTheLeft of
-                                Just (_ :: _) ->
-                                    Just
-                                        (Rule.errorWithFix
-                                            { message = "Appending literal lists could be simplified to be a single List"
-                                            , details = [ "Try moving all the elements into a single list." ]
-                                            }
-                                            checkInfo.fnRange
-                                            [ Fix.removeRange { start = (Node.range listArgToTheRight).end, end = checkInfo.parentRange.end }
-                                            , Fix.replaceRangeBy
-                                                { start = checkInfo.parentRange.start, end = startWithoutBoundary (Node.range listArgToTheRight) }
-                                                ("[" ++ checkInfo.extractSourceCode (rangeWithoutBoundaries (Node.range listArgToTheLeft)) ++ ",")
-                                            ]
-                                        )
+                    else
+                        case collection.fromListLiteralRange checkInfo.lookupTable secondArg_ of
+                            Just literalListRangeSecond ->
+                                case collection.fromListLiteralRange checkInfo.lookupTable checkInfo.firstArg of
+                                    Just literalListRangeFirst ->
+                                        Just
+                                            (Rule.errorWithFix
+                                                { message = qualifiedToString (qualify checkInfo.fn defaultQualifyResources) ++ " on literal " ++ collection.represents ++ "s can be turned into a single " ++ collection.represents
```

Comment:
"... on literal arrays can be turned into a single array" sounds a bit weird. Why not reuse the term literal? "can be turned into a single literal array"

Reply:
Sounds good :+1: